### PR TITLE
Added reporting script and updated privx_api module

### PR DESCRIPTION
**Pull request by kushns**
Added couple of reporting scripts 
1. get-access-report.py (Get historical connection data for ALL or individual user)
2. get-roles-mapping-report.py (Get role mapping data to generate access report, who can access what)

Added functions to privx_api module and updated _http_get

Updated README inside examples folder.

### CODE REVIEW

**Comment by jukeks** [comment]
Hey, good looking code! 

There are some minor style issues with the changes, some long lines, some misplaced whitespace etc. Minor things.

You should flake8 to check style issues by running `flake8` in the examples directory.  Also your function naming is not up to Python style guide: ["Function names should be lowercase, with words separated by underscores as necessary to improve readability."](https://www.python.org/dev/peps/pep-0008/#id45)

I would also refactor away all global keywords. They can be replaced by passing the variables as arguments to functions. Makes function easier to read.

File: privx_api/privx_api.py [hunk 1]
```diff
@@ -319,6 +322,16 @@ def get_sources(self) -> PrivXAPIResponse:
         """
         response = self._http_get("rolestore.sources")
         return PrivXAPIResponse(response, 200)
+        
+    def get_role_members(self, data: str) -> PrivXAPIResponse:
```
Comment (jukeks) [inline privx_api/privx_api.py:326]: I would name the argument as role_id.

File: examples/get-access-report.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+def ProcessError(messages):
+    print(messages)
+    sys.exit(2)
+    
+def main():
+    global user
+    if (len(sys.argv) > 3):
+        usage()
+        raise SystemExit
```
Comment (jukeks) [inline examples/get-access-report.py:103]: Perhaps rather use `sys.exit()` here to keep it consistent.

Comment (jukeks) [reply examples/get-access-report.py:103]: Or with non zero argument if this is an error case.

File: privx_api/privx_api.py (hunk 1 above)
Comment (kushns) [reply privx_api/privx_api.py]: argument name changed to role_id

File: examples/get-access-report.py (hunk 2 above)
Comment (kushns) [reply examples/get-access-report.py]: changed to sys.exit(2)

**Comment by kushns** [comment]
> 
> 
> Hey, good looking code!
> 
> There are some minor style issues with the changes, some long lines, some misplaced whitespace etc. Minor things.
> 
> You should flake8 to check style issues by running `flake8` in the examples directory. Also your function naming is not up to Python style guide: ["Function names should be lowercase, with words separated by underscores as necessary to improve readability."](https://www.python.org/dev/peps/pep-0008/#id45)
> 
> I would also refactor away all global keywords. They can be replaced by passing the variables as arguments to functions. Makes function easier to read.

Thanks for reviewing and providing input, all required change done.

File: examples/get-roles-mapping-report.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            name = role_data['name']
+            roles_data[name] = role_data['id']
+        return roles_data
+    else:
+        error = "Get Roles operation failed:"
+        process_error(error)
+
+
+def get_role_mapping_data(role_id, role_name):
+    global header_printed
```
Comment (jukeks) [inline examples/get-roles-mapping-report.py:36]: I think this is unnecessary now? It doesn't seem to be used in this function.

Comment (kushns) [reply examples/get-roles-mapping-report.py]: removed it now.

**Pull request merged**